src/Core/GeometryPrimitives/Tensor.cc

```diff
@@ -334,29 +339,65 @@ Vector Tensor::operator*(const Vector& v) const
 		v.x()*mat_[2][0]+v.y()*mat_[2][1]+v.z()*mat_[2][2]);
 }
 
+Tensor Tensor::operator/(const double s) const
+{
+  Tensor t1(*this);
+  for (int i=0; i<3; i++)
+    for (int j=0; j<3; j++)
+      t1.mat_[i][j]/=s;
+  if (t1.have_eigens_) {
+    t1.e1_/=s; t1.e2_/=s; t1.e3_/=s;
+    t1.l1_/=s; t1.l2_/=s; t1.l3_/=s;
+  }
+  return t1;
+}
+
 void Tensor::build_eigens_from_mat()
 {
   if (have_eigens_) return;
-  float ten[7];
-  ten[0] = 1.0;
-  ten[1] = mat_[0][0];
-  ten[2] = mat_[0][1];
-  ten[3] = mat_[0][2];
-  ten[4] = mat_[1][1];
-  ten[5] = mat_[1][2];
-  ten[6] = mat_[2][2];
-  float eval[3];
-  float evec[9];
-
-  tenEigensolve_f(eval, evec, ten);
-
-  e1_ = Vector(evec[0], evec[1], evec[2]);
-  e2_ = Vector(evec[3], evec[4], evec[5]);
-  e3_ = Vector(evec[6], evec[7], evec[8]);
-  l1_ = eval[0];
-  l2_ = eval[1];
-  l3_ = eval[2];
+  DenseMatrix dm = DenseMatrix(3, 3);
+  for (int i = 0; i < 3; ++i)
+    for (int j = 0; j < 3; ++j)
+      dm(i,j) = mat_[i][j];
+
+  auto es = Eigen::EigenSolver<Eigen::Matrix3d>(dm);
+  auto vecs = es.eigenvectors();
+  auto vals = es.eigenvalues();
+
+  e1_ = Vector(real(vecs(0, 0)), real(vecs(0, 1)), real(vecs(0, 2)));
+  e2_ = Vector(real(vecs(1, 0)), real(vecs(1, 1)), real(vecs(1, 2)));
+  e3_ = Vector(real(vecs(2, 0)), real(vecs(2, 1)), real(vecs(2, 2)));
+  l1_ = real(vals(0));
+  l2_ = real(vals(1));
+  l3_ = real(vals(2));
   have_eigens_ = 1;
+  reorderTensorValues();
+}
+
+void Tensor::reorderTensorValues()
+{
+  if (!have_eigens_) return;
+  std::vector<Vector> eigvecs = { e1_, e2_, e3_ };
+  std::vector<double> eigvals = { l1_, l2_, l3_ };
+  std::vector<std::pair<double, Vector>> sortList(3);
```

Comment:
```suggestion
std::map<double, Vector> sorted = { {e1_, l1_}, {e2_, l2_}, {e3_, l3_} };
```
Or why not just make a map, which will sort in the constructor

Reply:
neat